src/internal/CairoFrame.ts

```diff
@@ -0,0 +1,181 @@
+import { CamOverlayDrawingAPI, Align } from '../CamOverlayDrawingAPI';
+import ResourceManager from './ResourceManager';
+
+type RGB = [number, number, number];
+type RGBA = [number, number, number, number];
+type Options = {
+    x: number;
+    y: number;
+    width: number;
+    height: number;
+    bgImage?: string;
+    text?: string;
+    fontColor?: RGB;
+    bgColor?: RGBA;
+};
+type TMF = 'TFM_OVERFLOW' | 'TFM_SCALE' | 'TFM_TRUNCATE';
+type ObjectFitType = 'fill' | 'contain' | 'cover';
+
+export type DrawingCallback = (cod: CamOverlayDrawingAPI, cairo: string) => Promise<unknown>;
+export default class CairoFrame {
+    protected posX: number;
+    protected posY: number;
+    protected width: number;
+    protected height: number;
+
+    private text = '';
+    private align: Align = 'A_LEFT';
+    private textType: TMF = 'TFM_OVERFLOW';
+    private fontColor: RGB;
+    private fontName?: string;
+
+    private bgColor?: RGBA;
+    private bgImage?: string;
+    private bgType: ObjectFitType;
+
+    protected children = new Array<CairoFrame>();
+
+    constructor(opt: Options, protected rm: ResourceManager, private customDraw?: DrawingCallback) {
```

Comment:
nebylo by fajn udelat pro customDrawCallback i setter? Asi bych to prosel a vse co jde setnout pres options by melo jit na tom frame pozdeji i zmenit...